lxd/migrate.go

```diff
@@ -123,26 +105,20 @@ func (c *migrationFields) disconnect() {
 }
 
 func (c *migrationFields) sendControl(err error) {
-	message := ""
-	if err != nil {
-		message = err.Error()
-	}
+	c.controlLock.Lock()
+	defer c.controlLock.Unlock()
```

Comment:
Was there locking before? Why is there now?

Reply:
There was yes, it was calling c.send() which did the locking, ProtoSendControl itself doesn't lock so the lock had to be added here.

Reply:
Cool.